todo-list/src/components/List.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+    <div style="width: 300px">
+        <input v-model="newTodoName" placeholder="New Todo"
+               @keyup.enter="addNewTodo"/>
+        <button @click="addNewTodo" >Add</button>
+        <ListItem v-for="todo in todos" v-model="todos" :key="todo.id"
+                  :todo="todo"
+                  @on-delete-clicked="onTodoDelete"
+                  @on-name-changed="onNameChanged"
+                  @on-done-changed="onDoneChanged" >
+        </ListItem>
+    </div>
+</template>
+
+<script lang="ts">
+    import {Component, Vue} from "vue-property-decorator";
+    import ListItem from "@/components/ListItem.vue";
+    import {ToDo} from "@/todo";
+
+    @Component({components: {ListItem}})
+    export default class List extends Vue {
+        private newTodoName: string = "";
+        private todos: Array<ToDo> = [];
+
+        addNewTodo() {
+            let newTodo = new ToDo(this.newTodoName);
+            this.todos.push(newTodo);
+            this.newTodoName = "";
+
+            console.log(this.todos);
+        }
+
+        onTodoDelete(id: number, newName: string) {
+            for (let i = 0; i < this.todos.length; i++) {
```

Comment:
```suggestion
            for (let todo of this.todos) {
```
Maybe?

Reply:
To get the id - seems that is what you're looking for - you can do:
```suggestion
  this.todos.forEach((t, i) => {
```